src/worker.cc

```diff
@@ -429,6 +431,23 @@ void parse_location(valhalla::Location* location,
   if (!location->search_filter().has_max_road_class_case()) {
     location->mutable_search_filter()->set_max_road_class(valhalla::kMotorway);
   }
+
+  float waiting_secs = rapidjson::get<float>(r_loc, "/waiting", 0.f);
+  switch (location->type()) {
+    case Location_Type_kBreak:
+    case Location_Type_kBreakThrough:
+      // set waiting_time to 0 on origin/destination
+      {
+        auto loc_idx = location->correlation().original_index();
+        // TODO: waiting time can be less than 0
+        location->set_waiting_secs(loc_idx == 0 || is_last_loc || waiting_secs < 0.f ? 0.f
```

Comment:
We only set an actual `waiting_secs` time when it's
- not a origin
- not a destination
- and not negative